src/main/java/service/parse/DataParserImpl.java

```diff
@@ -0,0 +1,27 @@
+package service.parse;
+
+import model.FruitRecord;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataParserImpl implements DataParser {
+
+    @Override
+    public List<FruitRecord> parseFruitRecords(List<String> lines) {
+        List<FruitRecord> transactions = new ArrayList<>();
+
+        for (int i = 1; i < lines.size(); i++) {
+            String line = lines.get(i);
+            String[] parts = line.split(",");
```

Comment:
Consider validating the length of the 'parts' array to ensure that the line contains the expected number of elements before accessing them. This would prevent `ArrayIndexOutOfBoundsException` if the input data is malformed.